src/services/pact-verifier.service.ts

```diff
@@ -1,33 +1,35 @@
-import getPort from 'get-port';
 import { INestApplication, Inject, Injectable } from '@nestjs/common';
 import { Verifier } from '@pact-foundation/pact';
 import { PactModuleProviders } from '../common/pact-module-providers.enum';
 import { PactProviderOptions } from '../interfaces/pact-provider-module-options.interface';
 
 @Injectable()
 export class PactVerifierService {
-  public constructor(
-    @Inject(PactModuleProviders.ProviderOptions) private readonly options: PactProviderOptions,
-    @Inject(PactModuleProviders.PactVerifier) private readonly verifier: Verifier
-  ) {}
+  private verifier: Verifier | undefined = undefined;
+
+  public constructor(@Inject(PactModuleProviders.ProviderOptions) private readonly options: PactProviderOptions) {}
 
   public async verify(app: INestApplication): Promise<any> {
     const host = this.options.providerHost || 'localhost';
-    await app.listen(await getPort(), host);
+    const port = this.options.providerPort || 9123;
+
+    await app.listen(port, host);
 
     const appUrl = await app.getUrl();
 
-    const results = await this.verifier.verifyProvider({
-      ...this.options,
+    this.verifier = new Verifier({
```

Comment:
we need a new one here, as the v10 does not allow me to setthe URL without the new verifier